# **PERTEMUAN 1 - PENGENALAN ALGORITMA & PEMROGRAMAN DASAR**

# **LAPORAN PRAKTIKUM 1 STRUKTUR DATA**

*   Nama: Nadine Valia Azzahra
*   NIM: 2C2230007
*   PRODI: Sains Data

# **JUDUL PRAKTIKUM**

Pengenalan Algoritma Pemrograman Python: Percabangan (if-elif-else), Perulangan (for,
while), dan Logika Dasar dalam Struktur Data

# **TUJUAN PRAKTIKUM**

Setelah mengikuti praktikum ini, mahasiswa diharapkan mampu:

* Memahami dan menerapkan struktur percabangan (if, elif, else) dalam Python.
* Memahami dan menerapkan struktur perulangan (for) dengan range() dalam Python.
* Menggunakan operator aritmatika, logika, dan modulo untuk menyelesaikan persoalan.
* Membuat program interaktif yang menerima input dari pengguna.
* Menganalisis alur logika program Python sederhana.

# **DASAR TEORI**

* **Percabangan (Conditional Statement)**

Percabangan digunakan untuk mengeksekusi blok kode tertentu berdasarkan kondisi yang
dievaluasi. Python menggunakan kata kunci if, elif, dan else. Ekspresi kondisi dievaluasi
sebagai nilai boolean (True/False).

* **Perulangan (Loop)**

Perulangan digunakan untuk mengeksekusi blok kode berulang kali. Python menyediakan
for loop yang berjalan atas suatu iterable (seperti range), dan while loop yang berjalan
selama kondisi bernilai True. Kata kunci break digunakan untuk keluar dari loop,
sedangkan continue untuk melanjutkan ke iterasi berikutnya. Blok else pada loop
dieksekusi jika loop selesai tanpa break.

* **Operator**

Python mendukung berbagai operator: operator aritmatika (+, -, *, /, %), operator
perbandingan (==, !=, >, <, >=, <=), dan operator logika (and, or, not). Operator modulo
(%) sangat berguna untuk memeriksa habis-bagi suatu bilangan.

* **Input/Output**

Fungsi input() digunakan untuk menerima data dari pengguna dalam bentuk string. Fungsi
int() digunakan untuk mengonversi string menjadi integer. Fungsi print() digunakan untuk
menampilkan output ke layar, dengan kemampuan menampilkan beberapa argumen
sekaligus.

# **LANGKAH PRAKTIKUM**
Praktikum dilakukan dengan langkah-langkah berikut:

1. Praktikum dilakukan menggunakan Google Colab (colab.research.google.com) dan
membuat notebook baru.
2. Membaca dan memahami setiap soal tugas yang diberikan.
3. Menulis kode Python sesuai dengan logika yang dibutuhkan pada setiap tugas.
4. Menjalankan (run) setiap sel kode dan mengamati output yang dihasilkan.
5. Mencatat output hasil eksekusi program sebagai bukti hasil praktikum.
6. Menganalisis alur kerja program dan menyimpulkan pembelajaran dari setiap

# **TUGAS PRAKTIKUM**

In [ ]:
# Tugas 1. Menentukan positif, negatif, nol
bil = int(input("Masukkan bilangan: "))

if bil > 0:
  print("Bilangan positif")
elif bil < 0:
  print("Bilangan negatif")
else:
  print("Bilangan nol")

Masukkan bilangan: -100
Bilangan negatif


### **Analisis:**

Program menggunakan percabangan if-elif-else untuk membandingkan nilai bil dengan 0. Jika
bil > 0 maka positif, jika bil < 0 maka negatif, dan jika keduanya tidak terpenuhi maka nol.
Input -100 menghasilkan output 'Bilangan negatif' karena -100 < 0.

In [ ]:
# Tugas 2. Menampilkan bilangan genap 1-20'

for i in range(1,21):
  if i % 2 == 0:
    print(i)

2
4
6
8
10
12
14
16
18
20


## **Analisis:**

Program menggunakan for loop dengan range(1,21) untuk iterasi dari 1 hingga 20. Operator
modulo (%) digunakan untuk memeriksa sisa bagi; jika i % 2 == 0, berarti bilangan tersebut
genap. Output menampilkan 10 bilangan genap dari 2 hingga 20.

In [ ]:
# Tugas 3. Luas tanah (panjang 10, lebar 5)

panjang = 10
lebar = 5

luas = panjang * lebar

print("Luas tanah:", luas)

Luas tanah: 50


## **Analisis:**

Program menetapkan nilai panjang = 10 dan lebar = 5 sebagai variabel tetap, lalu menghitung
luas dengan rumus luas = panjang * lebar. Hasil perkalian 10 * 5 = 50 yang kemudian dicetak
sebagai output.

In [ ]:
# Tugas 4. Bilangan terbesar dari 3 input

a = int(input("Masukkan bilangan 1:"))
b = int(input("Masukkan bilangan 2:"))
c = int(input("Masukkan bilangan 3:"))

if a >= b and a >= c:
  print("Bilangan terbesar:", a)
elif b >= a and b >= c:
  print("Bilangan terbesar:", b)
else:
  print("Bilangan terbesar:", c)

Masukkan bilangan 1:6
Masukkan bilangan 2:9
Masukkan bilangan 3:0
Bilangan terbesar: 9


## **Analisis:**

Program menerima 3 input integer kemudian membandingkan menggunakan operator logika
AND. Kondisi pertama memeriksa apakah a >= b DAN a >= c. Jika tidak, memeriksa apakah
b >= a DAN b >= c. Jika keduanya tidak, maka c adalah yang terbesar. Dengan input 6, 9, 0.
nilai 9 (b) adalah yang terbesar.

In [ ]:
# Tugas 5. Menghitung rata-rata dari 5 nilai yang diinput user

n1 = int(input("Masukkan nilai 1: "))
n2 = int(input("Masukkan nilai 2:"))
n3 = int(input("Masukkan nilai 3: "))
n4 = int(input("Masukkan nilai 4:"))
n5 = int(input("Masukkan nilai 5: "))

rata = (n1 + n2 + n3 + n4 + n5) / 5
print("Rata-rata:", rata)

if rata >= 80:
  print("Baik")
elif rata >= 70:
  print("Cukup")
else:
  print("Kurang")

Masukkan nilai 1: 40
Masukkan nilai 2:90
Masukkan nilai 3: 80
Masukkan nilai 4:77
Masukkan nilai 5: 100
Rata-rata: 77.4
Cukup


## **Analisis:**

Program menerima 5 input nilai lalu menghitung rata-rata dengan rumus (n1+n2+n3+n4+n5)/5.
Dengan input 40, 90, 80, 77, 100 , rata-rata = 387/5 = 77.4. Karena 77.4 >= 70 tetapi < 80,
maka kategorinya adalah 'Cukup'.

In [ ]:
# Tugas 6. Diskon belanja

total = int(input("Masukkan total belanja:"))

if total >= 100000:
  diskon = total * 0.1
else:
  diskon = 0

total_bayar = total - diskon

print("Total belanja:", total_bayar)
print("Diskon", diskon)


Masukkan total belanja:455000
Total belanja: 409500.0
Diskon 45500.0


## **Analisis:**

Program memeriksa apakah total belanja >= 100.000. Jika iya, diskon dihitung sebesar 10%
dari total. Dengan input 455.000, diskon = 455.000 * 0.1 = 45.500, sehingga total_bayar =
455.000 - 45.500 = Rp. 409.500.

In [ ]:
# Tugas 7. Menampilkan tabel perkalian dari angka yang diinput user (1-10)

angka = int(input("Masukkan angka:"))

for i in range(1, 11):
  print(angka, "x", i, "=", angka * i)

Masukkan angka:20
20 x 1 = 20
20 x 2 = 40
20 x 3 = 60
20 x 4 = 80
20 x 5 = 100
20 x 6 = 120
20 x 7 = 140
20 x 8 = 160
20 x 9 = 180
20 x 10 = 200


## **Analisis:**

Program menggunakan for loop dengan range(1,11) untuk menghasilkan 10 iterasi (i = 1 sampai
10). Pada setiap iterasi, program mencetak format 'angka x i = hasil'. Dengan input 20, dicetak
tabel perkalian 20 dari 20x1 hingga 20x10.

In [ ]:
# Tugas 8. Menghitung faktorial dari suatu bilangan

bil = int(input("Masukkan bilangan: "))
faktorial = 1

for i in range(1, bil + 1):
  faktorial *= i


print("Hasil faktorial:", faktorial)

Masukkan bilangan: 8
Hasil faktorial: 40320


## **Analisis:**

Program dimulai dengan variabel faktorial = 1, lalu nilainya dikalikan secara bertahap dengan
angka dari 1 sampai bilangan yang dimasukkan. Misalnya jika input 8, perhitungannya menjadi
1 × 2 × 3 × 4 × 5 × 6 × 7 × 8 sehingga hasilnya 40320. Operator *= digunakan untuk menyingkat
penulisan, di mana faktorial *= i berarti faktorial = faktorial * i.  

In [ ]:
# Tugas 9. Menentukan apakah suatu bilangan termasuk bilangan prima atau bukan

bil = int(input("Masukkan bilangan: "))

if bil <= 1:
    print("Bukan bilangan prima")
else:
    for i in range(2, bil):
        if bil % i == 0:
            print("Bukan bilangan prima")
            break
    else:
        print("Bilangan prima")

Masukkan bilangan: 4
Bukan bilangan prima


### **Analisis:**

Program memakai for-else, di mana bagian else akan dijalankan jika perulangan selesai tanpa
break. Saat input 4, program mengecek 4 % 2 == 0 dan hasilnya benar, jadi langsung
menampilkan “Bukan bilangan prima” lalu berhenti karena break. Bilangan prima sendiri hanya
bisa dibagi oleh 1 dan dirinya sendiri.

In [ ]:
# Tugas 10. Membuat program login dengan ketentuan:
# Username = "admin", Password = "123"
# Maks percobaan = 3 kali
# Jika gagal 3 kali maka tampilan = "Akses Ditolak"

username_benar = "admin"
password_benar = "123"

for i in range(3):
    username = input("Masukan username: ")
    password = input("Masukan password: ")

    if username == username_benar and password == password_benar:
        print("Login berhasil")
        break
    else:
        print("Password salah, coba lagi")
else:
    print("Akses ditolak")

Masukan username: keren
Masukan password: 333
Password salah, coba lagi
Masukan username: sigma
Masukan password: 666
Password salah, coba lagi
Masukan username: windah
Masukan password: 111
Password salah, coba lagi
Akses ditolak


### **Analisis:**

Program menggunakan for-else dengan range(3) untuk memberi batas 3 kali percobaan login.
Jika username dan password benar, program menampilkan “Login berhasil” lalu berhenti
dengan break. Jika semua percobaan gagal dan tidak ada break, bagian else akan dijalankan dan
menampilkan “Akses ditolak”. Ini adalah contoh sederhana sistem autentikasi.